### PR TITLE
Add Rafter to installation process

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Add Rafter to installation process:
  - change installation yamls
  - add `rafter` folder in `resources` folder
  - add crds from `rafter` to `resources/cluster-essentials` Chart

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/6239
https://github.com/kyma-project/kyma/issues/6470

### CODE REVIEW

File: resources/rafter/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+- mutate the file
+- extract some of the file information and put it in the status of the custom resource
+
+Rafter comes with the following set of services and extensions compatible with Rafter webhooks:
+
+- [Upload Service](./charts/rafter-upload-service) (optional service)
+- [Front Matter Service](./charts/rafter-front-matter-service) (extension)
+- [AsyncAPI Service](./charts/rafter-asyncapi-service) (extension)
+
+To learn more about the Rafter, see the [Rafter repository](https://github.com/kyma-project/rafter).
```
Comment (kazydek) [inline resources/rafter/README.md]:
```suggestion
To learn more about the Rafter, see the [Rafter repository](https://github.com/kyma-project/rafter) and read the [Rafter documentation](https://kyma-project.io/docs/components/rafter)
```

File: resources/rafter/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+To learn more about the Rafter, see the [Rafter documentation](https://kyma-project.io/docs/components/rafter).
```
Comment (kazydek) [inline resources/rafter/README.md]:
```suggestion
```

File: resources/rafter/charts/minio/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# MinIO
+
+## Overview
+
+MinIO is an open source object storage server with Amazon S3 compatible API. Kyma provides MinIO as a core component to store static content. For example, documentation, images, or videos. The size of an object can range from a few KBs to a maximum of 5TB. In the long term, you can replace MinIO with an external solution, such as AWS S3.
```
Comment (kazydek) [inline resources/rafter/charts/minio/README.md]:
```suggestion
MinIO is an open-source object storage server with Amazon S3 compatible API. Kyma provides MinIO as a core component to store static content. For example, documentation, images, or videos. The size of an object can range from a few KBs to a maximum of 5TB. 
```

File: resources/rafter/charts/minio/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+To learn more about the MinIO, see the [documentation](https://docs.min.io/).
```
Comment (kazydek) [inline resources/rafter/charts/minio/README.md]:
```suggestion
To learn more about MinIO, see the [official documentation](https://docs.min.io/).
```

File: resources/rafter/charts/rafter-asyncapi-service/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Rafter AsyncAPI Service
+
+## Overview
+
+The Rafter AsyncAPI Service is an HTTP server used to process AsyncAPI specifications. It contains the `/validate` and `/convert` HTTP endpoints which accept `multipart/form-data` forms:
+
+- The `/validate` endpoint validates the AsyncAPI specification against the AsyncAPI schema in version 2.0.0., using the [AsyncAPI Parser](https://github.com/asyncapi/parser).
+- The `/convert` endpoint converts the version and format of the AsyncAPI files.
+
+To learn more about the Rafter AsyncAPI Service, see the [documentation](https://kyma-project.io/docs/components/rafter/#details-async-api-service).
```
Comment (kazydek) [inline resources/rafter/charts/rafter-asyncapi-service/README.md]:
```suggestion
To learn more about the Rafter AsyncAPI Service, read [this](https://kyma-project.io/docs/components/rafter/#details-asyncapi-service) documentation.
```

File: resources/rafter/charts/rafter-controller-manager/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Rafter Controller Manager
+
+## Overview
+
+The Rafter Controller Manager runs the AssetGroup, Asset, and Bucket Controllers that manage AssetGroup, Asset, and Bucket custom resources (CR).
+
+To learn more about the Rafter Controller Manager, see the [documentation](https://kyma-project.io/docs/components/rafter).
```
Comment (kazydek) [inline resources/rafter/charts/rafter-controller-manager/README.md]:
```suggestion
To learn more about the Rafter Controller Manager, see the [Rafter documentation](https://kyma-project.io/docs/components/rafter).
```

File: resources/rafter/charts/rafter-front-matter-service/README.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# Rafter Front Matter Service
+
+## Overview
+
+The Rafter Front Matter Service is an HTTP server used for extracting metadata from text files. It contains a simple HTTP endpoint which accepts `multipart/form-data` forms. The service extracts YAML front matter metadata from text files of all extensions.
```
Comment (kazydek) [inline resources/rafter/charts/rafter-front-matter-service/README.md]:
```suggestion
The Front Matter Service is an HTTP server used for extracting metadata from text files. It contains a simple HTTP endpoint which accepts `multipart/form-data` forms. The service extracts YAML front matter metadata from text files of all extensions.
```

File: resources/rafter/charts/rafter-asyncapi-service/README.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
+# Rafter AsyncAPI Service
+
+## Overview
+
+The Rafter AsyncAPI Service is an HTTP server used to process AsyncAPI specifications. It contains the `/validate` and `/convert` HTTP endpoints which accept `multipart/form-data` forms:
```
Comment (kazydek) [inline resources/rafter/charts/rafter-asyncapi-service/README.md]:
```suggestion
The AsyncAPI Service is an HTTP server used to process AsyncAPI specifications. It contains the `/validate` and `/convert` HTTP endpoints which accept `multipart/form-data` forms:
```

File: resources/rafter/charts/rafter-front-matter-service/README.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+
+To learn more about the Rafter Front Matter Service, see the [documentation](https://kyma-project.io/docs/components/rafter/#details-front-matter-service).
```
Comment (kazydek) [inline resources/rafter/charts/rafter-front-matter-service/README.md]:
```suggestion
To learn more about the Front Matter Service, see the [documentation](https://kyma-project.io/docs/components/rafter/#details-front-matter-service).
```

File: resources/rafter/charts/rafter-upload-service/README.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# Rafter Upload Service
+
+## Overview
+
+The Upload Service is an HTTP server used for hosting static files in [MinIO](https://min.io/). It contains a simple HTTP endpoint which accepts `multipart/form-data` forms. It uploads files to dedicated private and public system buckets that the service creates in MinIO, instead of Rafter. This service is particularly helpful if you do not have your own storage place from which Rafter could fetch assets. You can also use this service for development purposes to host files temporarily, without the need to rely on external providers.
+
+To learn more about the Rafter Upload Service, see the [documentation](https://kyma-project.io/docs/components/rafter/#details-upload-service).
```
Comment (kazydek) [inline resources/rafter/charts/rafter-upload-service/README.md]:
```suggestion
To learn more about the Upload Service, read [this](https://kyma-project.io/docs/components/rafter/#details-upload-service).
``` document

File: resources/rafter/charts/rafter-upload-service/README.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
+# Rafter Upload Service
+
+## Overview
+
+The Upload Service is an HTTP server used for hosting static files in [MinIO](https://min.io/). It contains a simple HTTP endpoint which accepts `multipart/form-data` forms. It uploads files to dedicated private and public system buckets that the service creates in MinIO, instead of Rafter. This service is particularly helpful if you do not have your own storage place from which Rafter could fetch assets. You can also use this service for development purposes to host files temporarily, without the need to rely on external providers.
+
+To learn more about the Upload Service, go to the [Rafter repository](https://github.com/kyma-project/rafter).
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply resources/rafter/charts/rafter-upload-service/README.md]: Changed to https://github.com/kyma-project/rafter/blob/master/cmd/uploader

File: resources/rafter/charts/rafter-front-matter-service/README.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
+# Rafter Front Matter Service
+
+## Overview
+
+The Front Matter Service is an HTTP server used for extracting metadata from text files. It contains a simple HTTP endpoint which accepts `multipart/form-data` forms. The service extracts YAML front matter metadata from text files of all extensions.
+
+To learn more about the Front Matter Service, go to the [Rafter repository](https://github.com/kyma-project/rafter).
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply resources/rafter/charts/rafter-front-matter-service/README.md]: Changed to https://github.com/kyma-project/rafter/blob/master/cmd/extension/frontmatter

File: resources/rafter/charts/rafter-controller-manager/README.md [hunk 13]
```diff
@@ -0,0 +1,7 @@
+# Rafter Controller Manager
+
+## Overview
+
+The Controller Manager runs the AssetGroup, Asset, and Bucket Controllers that manage AssetGroup, Asset, and Bucket custom resources (CR).
+
+To learn more about the Controller Manager, go to the [Rafter repository](https://github.com/kyma-project/rafter).
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply resources/rafter/charts/rafter-controller-manager/README.md]: Changed to https://github.com/kyma-project/rafter/blob/master/cmd/manager

File: resources/rafter/charts/rafter-asyncapi-service/README.md [hunk 14]
```diff
@@ -0,0 +1,10 @@
+# Rafter AsyncAPI Service
+
+## Overview
+
+The AsyncAPI Service is an HTTP server used to process AsyncAPI specifications. It contains the `/validate` and `/convert` HTTP endpoints which accept `multipart/form-data` forms:
+
+- The `/validate` endpoint validates the AsyncAPI specification against the AsyncAPI schema in version 2.0.0., using the [AsyncAPI Parser](https://github.com/asyncapi/parser).
+- The `/convert` endpoint converts the version and format of the AsyncAPI files.
+
+To learn more about the AsyncAPI Service, go to the [Rafter repository](https://github.com/kyma-project/rafter).
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply resources/rafter/charts/rafter-asyncapi-service/README.md]: Changed to https://github.com/kyma-project/rafter/blob/master/cmd/extension/asyncapi

File: resources/rafter/charts/rafter-upload-service/values.yaml [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+      pullPolicy: IfNotPresent
+    minio_client:
+      repository: 'minio/mc'
+      tag: 'RELEASE.2019-04-03T17-59-57Z'
+      pullPolicy: IfNotPresent
+  pre:
+    minioDeploymentRefName:
+    minioSecretRefName:
+  post:
+    minioSecretRefName: dupa
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply resources/rafter/charts/rafter-upload-service/values.yaml]: Removed dupa :P

File: resources/rafter/charts/rafter-asyncapi-service/templates/_helpers.tpl [hunk 16]
```diff
@@ -0,0 +1,79 @@
+{{/* vim: set filetype=mustache: */}}
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply resources/rafter/charts/rafter-asyncapi-service/templates/_helpers.tpl]: As I see, all helpers in charts in `helm/stable` have this line, so I prefer to stay with it.

File: resources/rafter/charts/rafter-asyncapi-service/README.md [hunk 17]
```diff
@@ -0,0 +1,10 @@
+# Rafter AsyncAPI Service
```
Comment (mmitoraj) [inline resources/rafter/charts/rafter-asyncapi-service/README.md]: I would suggest changing the chart name from `charts/rafter-asyncapi-service` to `charts/asyncapi-service`. What follows, please update the README title to `# AsyncAPI Service.`

File: resources/rafter/charts/rafter-front-matter-service/README.md [hunk 18]
```diff
@@ -0,0 +1,7 @@
+# Rafter Front Matter Service
```
Comment (mmitoraj) [inline resources/rafter/charts/rafter-front-matter-service/README.md]: I would suggest changing the chart name from `chart/rafter-front-matter-service` to `chart/front-matter-service`. What follows, please update the README title to `# Front Matter Service`.

File: resources/rafter/charts/rafter-upload-service/README.md [hunk 19]
```diff
@@ -0,0 +1,7 @@
+# Rafter Upload Service
```
Comment (mmitoraj) [inline resources/rafter/charts/rafter-upload-service/README.md]: I would suggest changing the chart name from `charts/rafter-upload-service` to `charts/upload-service`. What follows, please update the README title to `# Upload Service`.

File: resources/rafter/charts/rafter-asyncapi-service/README.md (hunk 17 above)
Comment (magicmatatjahu) [reply resources/rafter/charts/rafter-asyncapi-service/README.md]: Changed

File: resources/rafter/charts/rafter-front-matter-service/README.md (hunk 18 above)
Comment (magicmatatjahu) [reply resources/rafter/charts/rafter-front-matter-service/README.md]: Changed

File: resources/rafter/charts/rafter-upload-service/README.md (hunk 19 above)
Comment (magicmatatjahu) [reply resources/rafter/charts/rafter-upload-service/README.md]: Changed

File: resources/rafter/charts/controller-manager/charts/minio/templates/rafter-minio-envs.yaml [hunk 20]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "minio.fullname" . }}-envs
+data:
```
Comment (m00g3n) [inline resources/rafter/charts/controller-manager/charts/minio/templates/rafter-minio-envs.yaml]: Please add:
APP_INTERNAL_ENDPOINT: http://rafter-minio.kyma-system.svc.cluster.local:9000
and use it in rafter-upload-service deployment.

**Comment by m00g3n** [comment]
Awsome!

**Pull request merged**